nala/utils/base-reporter.js

```diff
@@ -0,0 +1,223 @@
+/* eslint-disable max-len, class-methods-use-this, no-empty-function, no-console */
+
+const { sendSlackMessage } = require('./slack.js');
+
+// Playwright will include ANSI color characters and regex from below
+// https://github.com/microsoft/playwright/issues/13522
+// https://github.com/chalk/ansi-regex/blob/main/index.js#L3
+
+const pattern = [
+  '[\\u001B\\u009B][[\\]()#;?]*(?:(?:(?:(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]+)*|[a-zA-Z\\d]+(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]*)*)?\\u0007)',
+  '(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PR-TZcf-nq-uy=><~]))',
+].join('|');
+
+const ansiRegex = new RegExp(pattern, 'g');
+
+// limit failed status
+const failedStatus = ['failed', 'flaky', 'timedOut', 'interrupted'];
+
+function stripAnsi(str) {
+  if (!str || typeof str !== 'string') return str;
+  return str.replace(ansiRegex, '');
+}
+
+class BaseReporter {
+  constructor(options) {
+    this.options = options;
+    this.results = [];
+    this.passedTests = 0;
+    this.failedTests = 0;
+    this.skippedTests = 0;
+  }
+
+  onBegin(config, suite) {
+    this.config = config;
+    this.rootSuite = suite;
+  }
+
+  async onTestEnd(test, result) {
+    const { title, retries, _projectId } = test;
+    const {
+      name, tags, url, browser, env, branch, repo,
+    } = this.parseTestTitle(title, _projectId);
+    const {
+      status,
+      duration,
+      error: { message: errorMessage, value: errorValue, stack: errorStack } = {},
+      retry,
+    } = result;
+
+    if (retry < retries && status === 'failed') {
+      return;
+    }
+    this.results.push({
+      title,
+      name,
+      tags,
+      url,
+      env,
+      browser,
+      branch,
+      repo,
+      status: failedStatus.includes(status) ? 'failed' : status,
+      errorMessage: stripAnsi(errorMessage),
+      errorValue,
+      errorStack: stripAnsi(errorStack),
+      stdout: test.stdout,
+      stderr: test.stderr,
+      duration,
+      retry,
+    });
+    if (status === 'passed') {
+      this.passedTests += 1;
+    } else if (failedStatus.includes(status)) {
+      this.failedTests += 1;
+    } else if (status === 'skipped') {
+      this.skippedTests += 1;
+    }
+  }
+
+  async onEnd() {
+    const summary = this.printResultSummary();
+    const resultSummary = { summary };
+
+    if (process.env.SLACK_WH) {
+      try {
+        await sendSlackMessage(process.env.SLACK_WH, resultSummary);
+      } catch (error) {
+        console.log('----Failed to publish result to slack channel----');
+      }
+    }
+  }
+
+  printResultSummary() {
+    const totalTests = this.results.length;
+    const passPercentage = ((this.passedTests / totalTests) * 100).toFixed(2);
+    const failPercentage = ((this.failedTests / totalTests) * 100).toFixed(2);
+    const miloLibs = process.env.MILO_LIBS || '';
```

Comment:
Same with the other process variables, have they been added to milo as well?

Reply:
> Same with the other process variables, have they been added to milo as well?
-> This process variable is set/updated as part of `run-nala-tests-on-milolibs` GitHub action flow,